prometheus/testutil/testutil.go

```diff
@@ -254,6 +254,9 @@ func compareMetricFamilies(got, expected []*dto.MetricFamily, metricNames ...str
 	if metricNames != nil {
 		got = filterMetrics(got, metricNames)
 		expected = filterMetrics(expected, metricNames)
+		if len(metricNames) > len(got) {
```

Comment:
I was thinking to use something like slices, to find the metricName that is not present in `got`. Would the additional package (slices is not yet in the std lib in go 1.19) and extra complexity of this function be outweighed by the improved usability. You would be able to see the misspelled or missing metricName in the error text.

Reply:
I added a commit. Finding the difference in 2 slices seems so easy but always requires so much code.

I added the "resulting" slice to the error with `%v`. I wonder if we should somehow allow callers of `compareMetricFamilies` to extract the missing metric Names? I guess that would require an API change :( 
It is just awkward to test if the "correct" metric names are in the error message.

Reply:
any feedback @bwplotka ?

Reply:
Why exact metric names would be useful? Isn't it enough to return error?

`compareMetricFamilies` has a clear semantics - we are just ensuring the error is more readable and comparison true.

I think small code complexity is fine here - proposed below what can be done.

Reply:
> Why exact metric names would be useful? Isn't it enough to return error?

right, error should be enough

Reply:
Isn't it better to check the actual metrics names are included in the given **metricsNames**? The reason is if we have a test table including several tests and say it has 3 metics but sometimes it may trigger only two for a specific test scenario. In that execution we check this comparison passing all three metrics names but with this change it breaks the flow from this point `if len(metricNames) > len(got) {`

Reply:
I think `if len(metricNames) < len(got)` it is ok to give an error, but if other way around need to check whether got names are matched with given names.

Reply:
This change breaks Mimir tests [when upgrading to v1.20.4](https://github.com/grafana/mimir/pull/9307), although I'm not sure exactly why yet.

Reply:
OK, I can see it breaks Mimir tests because they expect to be able to compare with a greater metrics set than what is present. After this change, that results in an error.

Reply:
This change breaks mongodb_exporter as well, we test there if metric doesn't appear